File: core/src/main/java/ru/hits/core/domain/dto/operation/OperationShortDTO.java
package ru.hits.core.domain.dto.operation;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;
import ru.hits.core.domain.enums.OperationTypeEnum;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@Schema(description = "DTO, представляющий операцию со счётом")
public class OperationShortDTO {

    @Schema(description = "Уникальный идентификатор операции", example = "a1234567-b89b-12d3-a456-426614174000")
    private UUID id;

    @Schema(description = "Сумма операции", example = "500.75")
    private Float amount;

    @Schema(description = "Значение определяющее была ли операция плюсом к счету или минусом", example = "true")
    private Boolean directionToMe;

    @Schema(description = "Дата и время операции", example = "2024-02-22T14:30:00Z")
    private Instant transactionDateTime;

    @Schema(description = "Тип операции", example = "WITHDRAWAL")
    private OperationTypeEnum operationType;
}

File: employee-app/HbankPRO/app/src/main/java/com/example/h_bankpro/data/dto/OperationShortDto.kt
package com.example.h_bankpro.data.dto

import com.example.h_bankpro.data.OperationType
import com.example.h_bankpro.domain.model.OperationShort
import kotlinx.datetime.Instant
import kotlinx.datetime.toJavaInstant

import kotlinx.serialization.Serializable
import java.time.ZoneId

@Serializable
data class OperationShortDto(
    val id: String,
    val amount: Float,
    val directionToMe: Boolean,
    val transactionDateTime: Instant,
    val operationType: OperationType
)

internal fun OperationShortDto.toDomain(): OperationShort =
    OperationShort(
        id = id,
        amount = amount,
        directionToMe = directionToMe,
        transactionDateTime = transactionDateTime.toJavaInstant()
            .atZone(ZoneId.systemDefault())
            .toLocalDateTime(),
        operationType = operationType
    )